packages/x-components/src/components/__tests__/location-injector.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { mount, Wrapper, VueClass } from '@vue/test-utils';
+import Vue from 'vue';
+import { Component, Inject } from 'vue-property-decorator';
+import { getDataTestSelector } from '../../__tests__/utils';
+import { FeatureLocation } from '../../types';
+import LocationInjector from '../location-injector.vue';
+
+@Component({
+  template: `<div data-test="inject"/>`
+})
+class Child extends Vue {
+  @Inject('location')
+  public location!: FeatureLocation;
+}
+
+/**
+ * Renders the `LocationInjector` component, exposing a basic API for testing.
+ *
+ * @param options - The options to render the component with.
+ * @returns The API for testing the `LocationInjector` component.
+ */
+function renderLocationInjector({
+  location
+}: RenderLocationInjectorOptions): RenderLocationInjectorAPI {
+  const wrapper = mount(
+    {
+      components: {
+        LocationInjector,
+        Child
+      },
+      template: `
+        <LocationInjector :location="location">
+          <Child />
+        </LocationInjector>`,
+      props: ['location']
+    },
+    {
+      propsData: {
+        location
+      }
+    }
+  );
+
+  return {
+    child: wrapper.find(getDataTestSelector('inject') as VueClass<Child>)
```

Comment:
There is something wrong with type inference and vue-test-utils. You should be able to write
```
    child: wrapper.findComponent(Child)
```

but type inference does not work and makes you explicitly type the generic:

```suggestion
    child: child: wrapper.findComponent<Child>(Child)
```

IMO this is easier than using a data-test selector and cast it. **Your way is fine**, I'm just saying this as free info.

Reply:
Nice one, thanks.